cmd/promtool/unittest.go

```diff
@@ -64,25 +72,30 @@ func RulesUnitTest(queryOpts promqltest.LazyLoaderOpts, runStrings []string, dif
 		}
 		fmt.Println()
 	}
+	err := junit.WriteXML(results)
+	if err != nil {
+		fmt.Fprintf(os.Stderr, "failed to write JUnit XML: %s\n", err)
```

Comment:
I get an error from this when not outputting results:

```
./promtool test rules cmd/promtool/testdata/unittest.yml
  SUCCESS

failed to write JUnit XML: invalid argument
```

Reply:
i did not find any error on my local , instead i got this as a xml output , 
<testsuite name="./testdata/unittest.yml" tests="5" failures="0" errors="0" skipped="0" timestamp="2024-07-24T12:08:45">
		<testcase name="unnamed#0"></testcase>
		<testcase name="unnamed#1"></testcase>
		<testcase name="unnamed#2"></testcase>
		<testcase name="unnamed#3"></testcase>
		<testcase name="unnamed#4"></testcase>
	</testsuite>

Reply:
The error is with the exact command I ran, if there's no junit output, it gives an error it can't write it.

The issue is actually https://github.com/prometheus/prometheus/pull/14506/files#diff-1bc6dfb35c5cfda81a13ff6d91a920ef951b34d72555ae5f51380fd5b7601b4eR383 -- that's a `**os.File`, so the check needs to be `*junitOutFile != nil`.